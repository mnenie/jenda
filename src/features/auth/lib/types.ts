export type Review = {
  id: string;
  author: string;
  text: string;
};
