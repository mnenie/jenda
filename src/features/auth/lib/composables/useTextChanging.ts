import { onActivated, ref } from 'vue';
import type { Ref } from 'vue';
import type { Review } from '../types';

export default function useTextChanging(reviews: Ref<Review[]>) {
  const currentIndex = ref(0);

  const changeReviewText = () => {
    currentIndex.value = Math.floor(Math.random() * reviews.value.length);
  };
  onActivated(() => {
    changeReviewText();
  });

  return {
    currentIndex,
    changeReviewText
  };
}
