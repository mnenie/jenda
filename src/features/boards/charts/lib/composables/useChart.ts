import { computed, effectScope, onScopeDispose, watchEffect } from 'vue';
import { useDark } from '@vueuse/core';
import type { ChartData, ChartOptions } from 'chart.js';

export default function useChart(data: ChartData, options: ChartOptions) {
  const isDark = useDark();

  const scope = effectScope();

  scope.run(() => {
    const bgColor = computed(() => {
      return isDark.value ? 'rgba(64, 64, 64, 0.9)' : 'rgb(228 228 231)';
    });
    const tickColor = computed(() => {
      return isDark.value ? 'rgb(163 163 163)' : 'rgb(82 82 82)';
    });

    watchEffect(() => {
      data.datasets[0].backgroundColor = bgColor.value;
      data.datasets[0].borderColor = bgColor.value;
      options.scales!.x!.ticks!.color = tickColor.value;
      options.scales!.y!.ticks!.color = tickColor.value;
      // @ts-expect-error missing type
      options.scales!.y!.border!.color = bgColor.value;
      // @ts-expect-error missing type
      options.scales!.x!.border!.color = bgColor.value;
    });
  });

  onScopeDispose(() => {
    scope.stop();
  });
}
