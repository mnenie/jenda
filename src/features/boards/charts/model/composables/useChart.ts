import { computed, effectScope, onScopeDispose, watch } from 'vue';
import { useDark } from '@vueuse/core';
import type { ChartData, ChartOptions } from 'chart.js';

export default function useChart(data: ChartData, options: ChartOptions) {
  const isDark = useDark();

  const scope = effectScope();

  scope.run(() => {
    const bgColor = computed(() => {
      return isDark.value ? 'rgba(64, 64, 64, 0.9)' : 'rgb(228 228 231)';
    });
    const tickColor = computed(() => {
      return isDark.value ? 'rgb(163 163 163)' : 'rgb(82 82 82)';
    });

    watch(
      [bgColor, tickColor],
      ([bg, tick], _) => {
        data.datasets[0].backgroundColor = bg;
        data.datasets[0].borderColor = bg;
        options.scales!.x!.ticks!.color = tick;
        options.scales!.y!.ticks!.color = tick;
        // @ts-expect-error missing type
        options.scales!.y!.border!.color = bg;
        // @ts-expect-error missing type
        options.scales!.x!.border!.color = bg;
      },
      { immediate: true }
    );
  });

  onScopeDispose(() => {
    scope.stop();
  });
}
