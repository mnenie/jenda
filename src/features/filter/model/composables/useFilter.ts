import { useEventListener } from '@vueuse/core';
import { nextTick } from 'vue';
import type { Ref } from 'vue';

export default function useFilter(
  inputRef: Ref<HTMLElement | null>,
  props: { isExpanded: boolean },
  emit: (e: 'onToggle') => void
) {
  const onToggleArea = () => {
    if (!props.isExpanded) {
      emit('onToggle');

      nextTick(() => {
        inputRef.value = document.getElementById('input') as HTMLInputElement;
        if (inputRef.value) {
          inputRef.value.focus();
        }
      });
    }
  };
  const handleKeyDownFilter = (e: KeyboardEvent) => {
    if (document) {
      inputRef.value = document.getElementById('input')!;
      if ((e.metaKey && e.key === 'k') || (e.ctrlKey && e.key === 'k')) {
        inputRef.value.focus();
      }
    }
  };

  useEventListener('keydown', handleKeyDownFilter);

  return {
    handleKeyDownFilter,
    onToggleArea
  };
}
