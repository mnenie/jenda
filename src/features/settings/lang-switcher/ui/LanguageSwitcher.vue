<script setup lang="ts">
import { UiSelect } from '@/shared/ui';
import { ref, shallowReactive } from 'vue';

// after -> locales instead
const options = shallowReactive(['English', 'Русский', '简体中文']);

const selectLang = ref('English');
</script>

<template>
  <UiSelect v-model="selectLang" :options />
</template>
