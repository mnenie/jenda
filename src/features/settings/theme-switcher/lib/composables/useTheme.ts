import { ref } from 'vue';
import type { Ref } from 'vue';
import { useColorMode } from '@vueuse/core';
import type { ThemeBlock } from '../../model';

export default function useTheme(themeBlocks: Ref<ThemeBlock[]>) {
  const { store } = useColorMode();

  const selectedThemeId = ref(themeBlocks.value.find((item) => item.active)?.id || 0);

  const changeActiveTheme = (id: number) => {
    themeBlocks.value = themeBlocks.value.map((theme, index) => ({
      ...theme,
      active: index === id
    }));
    selectedThemeId.value = id;
  };

  const onNewPreferences = () => {
    const selectedTheme = themeBlocks.value.find((item) => item.id === selectedThemeId.value);
    if (selectedTheme) {
      store.value = selectedTheme.value;
    }
    console.log(store.value);
  };

  return {
    changeActiveTheme,
    onNewPreferences
  };
}
