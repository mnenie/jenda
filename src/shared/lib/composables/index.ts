export * from './useLanguage';
export * from './useScroll';
export * from './useExpanded';
export * from './useLayoutPaths';
