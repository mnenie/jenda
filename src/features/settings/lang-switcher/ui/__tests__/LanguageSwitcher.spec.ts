import { mount } from '@vue/test-utils';
import { describe, it, expect, vi } from 'vitest';
import { nextTick, ref, shallowReactive } from 'vue';
import '@/shared/lib/vitest-utils/cookiesI18n-mock';
import i18n from '@/shared/lib/i18n';
import LanguageSwitcher from '../LanguageSwitcher.vue';
import { useLanguage } from '@/shared/lib/composables';
import { useI18n } from 'vue-i18n';

vi.mock('vue-i18n', async (importOriginal) => {
  const actual = await importOriginal();
  return {
    ...(actual as unknown as object),
    useI18n: () => ({
      locale: ref('en-US')
    })
  };
});

describe('tests for LanguageSwitcher.vue', () => {
  const wrapper = mount(LanguageSwitcher, {
    global: {
      plugins: [i18n]
    }
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should render subcomponents', () => {
    expect(wrapper.findComponent({ name: 'UiSelect' }).exists()).toBe(true);
  });
});

describe('useLanguage composable tests', () => {
  // mocks
  const options = shallowReactive([
    { name: 'English', value: 'en-US' },
    { name: 'Русский', value: 'ru-RU' },
    { name: '简体中文', value: 'zh-CN' }
  ]);

  it('should correctly sync language with `locale`', async () => {
    const { locale } = useI18n();
    const language = ref('');

    useLanguage(options, language);

    expect(language.value).toBe('English');

    locale.value = 'ru-RU';
    await nextTick();
    expect(locale.value).toBe('ru-RU');
  });
});
