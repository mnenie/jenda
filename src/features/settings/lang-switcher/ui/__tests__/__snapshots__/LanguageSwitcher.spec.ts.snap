// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for LanguageSwitcher.vue > should render correctly 1`] = `
"<div class="custom_select">
  <div class="selected"><span class="text-sm">English</span><svg xmlns="http://www.w3.org/2000/svg" width="15" height="15" viewBox="0 0 24 24" fill="none" stroke="var(--zinc-400)" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-chevrons-up-down-icon">
      <path d="m7 15 5 5 5-5"></path>
      <path d="m7 9 5-5 5 5"></path>
    </svg></div>
  <transition-stub name="dropdown" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`tests for LanguageSwitcher.vue > should render subcomponents 1`] = `
"<div class="custom_select">
  <div class="selected"><span class="text-sm">English</span><svg xmlns="http://www.w3.org/2000/svg" width="15" height="15" viewBox="0 0 24 24" fill="none" stroke="var(--zinc-400)" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-chevrons-up-down-icon">
      <path d="m7 15 5 5 5-5"></path>
      <path d="m7 9 5-5 5 5"></path>
    </svg></div>
  <transition-stub name="dropdown" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
</div>"
`;
